HotelBooking/HotelBooking.DataAccessLayer/Database/HotelContext.cs

```diff
@@ -0,0 +1,114 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace HotelBooking.DataAccessLayer.Database;
+
+public class HotelContext : DbContext
+{
+    public HotelContext()
+    {
+
+    }
+
+    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+    {
+
+        optionsBuilder.UseSqlServer(@"Server=localhost,1433;Database=HotelBooking;Trusted_Connection=True;");
+
+    }
+
+    public DbSet<Dictionary<string, object>> Guests => Set<Dictionary<string, object>>("Guest");
+    public DbSet<Dictionary<string, object>> Reservations => Set<Dictionary<string, object>>("Reservation");
+    public DbSet<Dictionary<string, object>> Rooms => Set<Dictionary<string, object>>("Room");
+
+    protected override void OnModelCreating(ModelBuilder modelBuilder)
+    {
+        modelBuilder.Entity<Guest>(
+            g =>
+            {
+                g.Property(g => g.GuestId)
+                    .HasColumnName("guest_id")
+                    .HasColumnType("int");
+                g.Property(g => g.Firstname)
+                    .HasColumnName("firstname")
+                    .HasColumnType("string");
+                g.Property(g => g.Lastname)
+                    .HasColumnName("lastname")
+                    .HasColumnType("string");
+                g.Property(g => g.Email)
+                    .HasColumnName("email")
+                    .HasColumnType("string");
+                g.Property(g => g.PhoneNumber)
+                    .HasColumnName("phone_number")
+                    .HasColumnType("string");
+                g.Property(g => g.DateBirth)
+                    .HasColumnName("date_birth")
+                    .HasColumnType("datetime");
+            });
+
+        modelBuilder.Entity<Reservation>(
```

Comment:
Analogicznie jak z gościem. Tutaj jednak zostawiłbym HasColumnType przy właściwości Price. Tutaj wydaje mi się że EF domyślnie by zmapował to na jakiś typ całkowitoliczbowy, a my chcemy z przecinkiem.
Trzeba właśnie uważać na typy danych przy mapowaniu, typy C#owe to jednak coś innego niż typy MSSQL :)

Reply:
![obraz](https://github.com/AnnaMalgorzata/Hotel-Booking/assets/132166200/f783059d-f3dd-4d73-a76c-414c07c4a84b)